janusgraph-core/src/main/java/org/janusgraph/graphdb/util/SubqueryIterator.java

```diff
@@ -73,11 +74,8 @@ public SubqueryIterator(JointIndexQuery.Subquery subQuery, IndexSerializer index
 
     @Override
     public boolean hasNext() {
-        if (!elementIterator.hasNext() && currentIds != null) {
```

Comment:
What happened to the `currentIds != null` check here? Before your change, the `SubQueryIterator` was also stopped if `currentIds` was `null`, even if the `elementIterator` had a next element.

Reply:
You moved the lines inside this `if` statement to a separate function named `close` and now call this function instead of executing the lines directly. But the condition inside the `if` has changed. It was
```java
if (!elementIterator.hasNext() && currentIds != null) {
   ...
}
```
before and now it is
```java
if (!elementIterator.hasNext()) {
   ...
}
```

You removed the ``&& currentIds != null` part.

Reply:
Got your question. What confused me was the sentence:

> Before your change, the SubQueryIterator was also stopped if currentIds was null, even if the elementIterator had a next element.

Anyway, I removed this `currentIds != null` condition because it is not needed anymore. In the old logic, it is needed, because `currentIds == null` means cached response is used, in which case there is no timer running & no results fetched from backend. 

After my changes, I moved the logic into the `close()` method. In that method, nothing would be done if `isTimerRunning` evaluates to false (note that `isTimerRunning == true` and `currentIds != null` are equivalent), so the effect remains the same.